### PR TITLE
[EngSys] adding ci checks for performance

### PR DESCRIPTION
**Pull request by seankane-msft**
* Runs `go build` and `go vet` on the testdata/perf directory if it exists

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -61,6 +61,14 @@ steps:
       PROXY_CERT: $(Build.SourcesDirectory)/eng/common/testproxy/dotnet-devcert.crt
       ${{ insert }}: ${{ parameters.EnvVars }}
 
+  - tast: PowerShell@2
+    display: 'Build Performance Tests'
+    inputs:
+      targetType: 'filePath'
+      filePath: ./eng/scripts/run_tests.ps1
```
Comment (benbp) [inline eng/pipelines/templates/steps/build-test.yml:68]: Is this supposed to be `Build_Perf.ps1`?

File: eng/scripts/Build_Perf.ps1 [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#Requires -Version 7.0
+
+Param(
+    [string] $serviceDirectory,
```
Comment (benbp) [inline eng/scripts/Build_Perf.ps1:4]:
```suggestion
    [string] $serviceDirectory
```

File: eng/scripts/Build_Perf.ps1 [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Push-Location sdk/$serviceDirectory
+
+if (Test-Path -Path testdata/perf) {
+    Write-Host "##[command] Building and vetting performance tests in sdk/$serviceDirectory/testdata/perf"
+
+    Write-Host "##[command] Executing 'go build .' in sdk/$serviceDirectory/testdata/perf"
+    go build .
+    if ($LASTEXITCODE) {
+        exit $LASTEXITCODE
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply eng/scripts/Build_Perf.ps1:15]: I see what you're pointing out, adding another `Pop-Location` before the exit calls

File: eng/scripts/Build_Perf.ps1 [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    [string] $serviceDirectory
+)
+
+Push-Location sdk/$serviceDirectory
+
+if (Test-Path -Path testdata/perf) {
+    Write-Host "##[command] Building and vetting performance tests in sdk/$serviceDirectory/testdata/perf"
+
+    Write-Host "##[command] Executing 'go build .' in sdk/$serviceDirectory/testdata/perf"
+    go build .
```
Comment (benbp) [inline eng/scripts/Build_Perf.ps1:13]: The go build will run in the service directory root. Is this intended or did you want another push location here into testdata/perf?

File: eng/scripts/Build_Perf.ps1 [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+$failed = $false
+
+foreach ($perfDir in $perfDirectories) {
+    Push-Location $perfDir
+
+    if (Test-Path -Path perf) {
+        Push-Location perf
+        Write-Host "##[command] Building and vetting performance tests in $perfDir/testdata/perf"
```
Comment (benbp) [inline eng/scripts/Build_Perf.ps1:24]:
```suggestion
        Write-Host "##[command] Building and vetting performance tests in $perfDir/perf"
```

File: eng/scripts/Build_Perf.ps1 [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        Write-Host "##[command] Executing 'go build .' in $perfDir/testdata/perf"
```
Comment (benbp) [inline eng/scripts/Build_Perf.ps1:26]:
```suggestion
        Write-Host "##[command] Executing 'go build .' in $perfDir/perf"
```

File: eng/scripts/Build_Perf.ps1 [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        go build .
+        if ($LASTEXITCODE) {
+            $failed = $true
+        }
+
+        Write-Host "##[command] Executing 'go vet .' in $perfDir/testdata/perf"
```
Comment (benbp) [inline eng/scripts/Build_Perf.ps1:32]:
```suggestion
        Write-Host "##[command] Executing 'go vet .' in $perfDir/perf"
```

**Review by benbp — APPROVED** [review]

**Pull request merged**